src/esp32/src/flash.c

```diff
@@ -4,16 +4,148 @@
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  */
 
-#include "flash.h"
+#include <target/flash.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <log.h>
+#include <private/rom.h>
+#include <private/soc_utils.h>
+
+#include "soc/gpio_reg.h"
```

Comment:
Do we need this header just for `GPIO_STRAP_REG` address?  If so we can drop that file and simply add 
```c
#define GPIO_STRAP_REG          (DR_REG_GPIO_BASE + 0x0038)
```

Reply:
I'm also in doubt when we should take the entire generated files from IDF, and when we can do with a couple of constants. Since gpio_reg.h is still not used elsewhere, let's stay with the simple